File: apps/backend/src/portfolio/AssetDividendRecord.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  JoinColumn,
} from 'typeorm';
import { PortfolioAsset } from './PortfolioAsset';
import { UUID } from '@mid/shared';
import { Account } from '../account/Account';

@Entity()
export class AssetDividendRecord {
  @PrimaryGeneratedColumn('uuid')
  id: UUID;

  @ManyToOne(() => PortfolioAsset, (portfolioAsset) => portfolioAsset.records, {
    nullable: false,
    eager: true,
  })
  @JoinColumn({ name: 'portfolio_asset_id' })
  portfolioAsset: PortfolioAsset;

  @Column({ type: 'bigint' })
  amount: number;

  @Column({ type: 'date' })
  dividendDate: Date;

  @ManyToOne(() => Account, {
    lazy: false,
    nullable: false,
  })
  @JoinColumn({ name: 'account_id' })
  account: Account;
}

File: backend/src/main/kotlin/com/kyoulho/mid/portfolio/AssetDividendRecord.kt
package com.kyoulho.mid.portfolio

import com.kyoulho.mid.account.entity.Account
import jakarta.persistence.*
import java.util.*

@Entity
data class AssetDividendRecord(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "portfolio_asset_id", nullable = false)
    val portfolioAsset: PortfolioAsset,

    @Column(nullable = false)
    val amount: Long,

    @Column(nullable = false)
    val dividendDate: Date,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "account_id", nullable = false)
    val account: Account
)
